runHarness/WorkloadDrivers/AuctionWorkloadDriver.pm

```diff
@@ -1174,19 +1186,87 @@ sub startRun {
 	my $endRunStatus = "";
 	my $endRunStatusRaw = "";
 	$url = $self->getControllerURL() . "/run/$runName/state";
+	my @curIntervalNames;
+	my @lastIntervalNames;
+	my @statsRunning;
+
 	while (!$runCompleted) {
 		$logger->debug("Sending get to $url");
 		$req = HTTP::Request->new( GET => $url );
 		$res = $ua->request($req);
 		$logger->debug(
 			"Response status line: " . $res->status_line . " for url " . $url );
 		if ( $res->is_success ) {
-			$endRunStatus = $json->decode( $res->content );	
+			$logger->debug("Response content: " . $res->content );
+			$endRunStatus = $json->decode( $res->content );
+
 			my $workloadStati = $endRunStatus->{'workloadStati'};
 			foreach my $workloadStatus (@$workloadStati) {
 				my $wkldName = $workloadStatus->{'name'};
-				my $curIntervalName = $workloadStatus->{'curInterval'}->{'name'};
-				$logger->debug("Workload $workloadNum, AppInstance: $wkldName: curInterval = $curIntervalName");
+				my $curInterval = $workloadStatus->{'curInterval'};
+				my $curIntervalName;
+				if ($curInterval) {
+					$curIntervalName = $curInterval->{'name'};
+					$logger->debug("Workload $workloadNum, AppInstance: $wkldName: curInterval = $curIntervalName");
+				}
+				if ($usingFindMaxLoadPathType) {
+					$wkldName =~ /appInstance(\d+)/;
+					my $appInstanceNum = $1;
+
+					if (defined $statsRunning[$appInstanceNum] && $statsRunning[$appInstanceNum]) {
+						$statsRunning[$appInstanceNum] = 0;
+						$console_logger->info("   [$wkldName] Stopping performance statistics on workload.");
+						$self->workload->stopStatsCollection($tmpDir);
```

Comment:
Just trying to understand what is happening here.  Are you calling startStats at the start, and stopStats at the end, of each VERIFYMAX iteration?  If so, I like that approach.

Reply:
Basically.  Calling startStats when the poll sees a new VERIFYMAX.  Calling stopStats one poll later (so 60 seconds).  Happens for each VERIFYMAX interation, so each of start and stop will be called at least twice.

Reply:
Sounds good to me.